File: Binary-search/Binary_search.py
def binarysearh(lst, x, left, right):
	if left <= right:
		mid = left + (right - left) // 2
		if lst[mid] == x:
			return mid
		elif lst[mid] > x:
			return binarysearh(lst, x, left, mid-1)
		else:
			return binarysearh(lst, x, mid+1, right)
	else:
		return -1


lst = [7, 5, 3, 6, 9, 1, 0, 2, 4, 8]
x = 5
print(binarysearh(lst, x, 0, len(lst)-1))

File: Searching algorithms/Binary-search/Binary_search.cpp
#include <iostream>
using namespace std;

int binarysearch(int arr[], int x, int left, int right)
{
	if (left <= right) {
		int mid = left + (right - left) / 2;
		if (arr[mid] == x) {
			return mid;
		}
		else if (arr[mid] > x) {
			return binarysearch(arr, x, left, mid-1);
		}
		else {
			return binarysearch(arr, x, mid+1, right);
		}
	}
	else {
		return -1;
	}
}


int main()
{
	int arr[] = {7, 5, 3, 6, 9, 8, 4, 2, 1, 0};
	int x = 3;
	int len = sizeof(arr) / sizeof(arr[0]);
	cout<<binarysearch(arr, x, 0, len)<<endl;
	return 0;
}